src/viam/sdk/rpc/server.cpp

```diff
@@ -47,8 +47,8 @@ void Server::add_resource(std::shared_ptr<Resource> resource) {
                << " but no matching resource server as found";
         throw std::runtime_error(buffer.str());
     }
-    auto resource_server = managed_servers_.at(std::move(api));
-    resource_server->resource_manager()->add(resource->name(), std::move(resource));
+    auto resource_server = managed_servers_.at(api);
+    resource_server->resource_manager()->add(resource->name(), resource);
```

Comment:
Surprising. It should be fine to move `resource` here.

Reply:
(per error message) `add` takes a `const&` so no move will happen. But that seems a bit strange to me. I've updated for `add` to take by value and consistently move.